### PR TITLE
Ssks: Update side distance

### PR DESCRIPTION
**Pull request by TruongQuangSB**
- The find geometry for GEO_Kante process will execute after the model complete load and run in background (like find BankingLine).
- Add distance from Signal to another track, when it is relevant.
- When open Ssks Table before the find geometry process complete, then hourglass will fill like BankingValue. But the table will be only refresh after all new value is set (by Bankvalue is the table refresh after a value found).

### CODE REVIEW

File: java/bundles/org.eclipse.set.ppmodel.extensions/src/org/eclipse/set/ppmodel/extensions/geometry/GEOKanteGeometryExtensions.java [hunk 1]
```diff
@@ -0,0 +1,373 @@
...
+	// Configuration options for Clothoids
+	private static final int CLOTHOID_SEGMENTS_MIN = 10;
+	private static final double CLOTHOID_SEGMENTS_PER_LENGTH = 0.5;
+	private static final int CLOTHOID_PRECISION = 5;
+	private GeometryFactory geometryFactory;
+	private Thread findGeometryThread;
+
+	private static boolean isFindProcessComplete = false;
+
+	private static Map<GEO_Kante, LineString> edgeGeometry;
```
Comment (TruongQuangSB) [inline java/bundles/org.eclipse.set.ppmodel.extensions/src/org/eclipse/set/ppmodel/extensions/geometry/GEOKanteGeometryExtensions.java:82]: To improve preformance, should find geometry for geokante process execute only one time

File: java/bundles/org.eclipse.set.ppmodel.extensions/src/org/eclipse/set/ppmodel/extensions/geometry/GEOKanteGeometryExtensions.java [hunk 2]
```diff
@@ -0,0 +1,373 @@
...
+				}
+			}, ToolboxConstants.CacheId.GEOKANTE_GEOMETRY);
+			findGeometryThread.start();
+		}
+
+		if (topic.equals(Events.CLOSE_SESSION) && findGeometryThread.isAlive()
+				&& !findGeometryThread.isInterrupted()) {
+			findGeometryThread.interrupt();
+		}
+	}
```
Comment (TruongQuangSB) [inline java/bundles/org.eclipse.set.ppmodel.extensions/src/org/eclipse/set/ppmodel/extensions/geometry/GEOKanteGeometryExtensions.java:121]: like find BankingLine process, this process will be excecute in background after model complete load

File: java/bundles/org.eclipse.set.ppmodel.extensions/src/org/eclipse/set/ppmodel/extensions/geometry/GEOKanteGeometryExtensions.java [hunk 3]
```diff
@@ -0,0 +1,373 @@
...
+	private static final int BLOSS_PRECISION = 20;
+
+	// Configuration options for Clothoids
+	private static final int CLOTHOID_SEGMENTS_MIN = 10;
+	private static final double CLOTHOID_SEGMENTS_PER_LENGTH = 0.5;
+	private static final int CLOTHOID_PRECISION = 5;
+	private GeometryFactory geometryFactory;
+	private Thread findGeometryThread;
+
+	private static boolean isFindProcessComplete = false;
```
Comment (pstuecker) [inline java/bundles/org.eclipse.set.ppmodel.extensions/src/org/eclipse/set/ppmodel/extensions/geometry/GEOKanteGeometryExtensions.java:80]: This shouldn't be static, as this is a service.

File: java/bundles/org.eclipse.set.ppmodel.extensions/src/org/eclipse/set/ppmodel/extensions/geometry/GEOKanteGeometryExtensions.java [hunk 4]
```diff
@@ -0,0 +1,373 @@
...
+	}
+
+	@Override
+	public void handleEvent(final Event event) {
+		final String topic = event.getTopic();
+		if (topic.equals(Events.TOPMODEL_CHANGED) && event.getProperty(
+				IEventBroker.DATA) instanceof final PlanPro_Schnittstelle schnitstelle) {
+			geometryFactory = new GeometryFactory();
+			createEdgeGeometryInstance();
+			findGeometryThread = new Thread(() -> {
```
Comment (pstuecker) [inline java/bundles/org.eclipse.set.ppmodel.extensions/src/org/eclipse/set/ppmodel/extensions/geometry/GEOKanteGeometryExtensions.java:100]: isFindProcessComplete needs to be set to false again

File: java/bundles/org.eclipse.set.feature.table.pt1/src/org/eclipse/set/feature/table/pt1/ssks/SsksTransformator.xtend [hunk 5]
```diff
@@ -818,25 +905,70 @@ class .simpleName»: «e.message» - failed to transform table contents''', e)
 					signalBefestigungAllg.befestigungArt.wert ==
 						ENUM_BEFESTIGUNG_ART_REGELANORDNUNG_MAST_NIEDRIG
 			]
-		].filterNull.map[singlePoints].flatten.forEach [ p |
+		].filterNull.map[singlePoints].flatten.toSet.forEach [ p |
 			val seitlicherAbstand = Math.round(
 				p.seitlicherAbstand.wert.doubleValue * 1000)
 			val wirkrichtung = signal.getWirkrichtung(p.topKante)
+
+			val distanceFromPoint = 7000 - Math.abs(seitlicherAbstand)
```
Comment (pstuecker) [inline java/bundles/org.eclipse.set.feature.table.pt1/src/org/eclipse/set/feature/table/pt1/ssks/SsksTransformator.xtend:913]: Should be a constant

**Review by pstuecker — CHANGES_REQUESTED** [review]

File: java/bundles/org.eclipse.set.feature.siteplan/src/org/eclipse/set/feature/siteplan/trackservice/GEOKanteMetadata.xtend [hunk 6]
```diff
@@ -69,7 +70,7 @@ class GEOKanteMetadata {
 		this.geoKante = geoKante
 		this.start = start
 		this.length = length
-		this.geometry = geoKante.geometry
+		this.geometry = geoKante.getGeometry
```
Comment (pstuecker) [inline java/bundles/org.eclipse.set.feature.siteplan/src/org/eclipse/set/feature/siteplan/trackservice/GEOKanteMetadata.xtend:73]:
```suggestion
		this.geometry = geoKante.geometry
```

File: java/bundles/org.eclipse.set.feature.siteplan/src/org/eclipse/set/feature/siteplan/trackservice/GEOKanteMetadata.xtend [hunk 7]
```diff
@@ -160,7 +161,7 @@ class GEOKanteMetadata {
 		this.start = start
 		this.length = (geoKante.GEOKanteAllg?.GEOLaenge?.wert ?:
 			BigDecimal.ZERO).doubleValue
-		this.geometry = geoKante.geometry
+		this.geometry = geoKante.getGeometry
```
Comment (pstuecker) [inline java/bundles/org.eclipse.set.feature.siteplan/src/org/eclipse/set/feature/siteplan/trackservice/GEOKanteMetadata.xtend:164]:
```suggestion
		this.geometry = geoKante.geometry
```

File: java/bundles/org.eclipse.set.core.services/src/org/eclipse/set/core/services/cache/NoCacheService.java [hunk 8]
```diff
@@ -29,4 +29,9 @@ public Cache getCache(final String cacheId, final String containerCacheId)
 			throws IllegalArgumentException {
 		return new NoCache();
 	}
+
+	@Override
+	public Boolean existCache(final String cacheID) {
+		throw new UnsupportedOperationException();
```
Comment (pstuecker) [inline java/bundles/org.eclipse.set.core.services/src/org/eclipse/set/core/services/cache/NoCacheService.java:35]:
```suggestion
		return Boolean.FALSE;
```

File: java/bundles/org.eclipse.set.feature.siteplan/src/org/eclipse/set/feature/siteplan/transform/RouteTransformator.xtend [hunk 9]
```diff
@@ -50,9 +51,10 @@ class RouteTransformator extends BaseTransformator<Strecke> {
 		shape = TrackTransformator.transformGeoForm(
 			geoKante.GEOKanteAllg.GEOForm)
 		val crs = geoKante.CRS
-		positions.addAll(geoKante.geometry.coordinates.map [
-			positionService.transformCoordinate(it, crs)
-		])
+		positions.addAll(geoKante.getGeometry.
```
Comment (pstuecker) [inline java/bundles/org.eclipse.set.feature.siteplan/src/org/eclipse/set/feature/siteplan/transform/RouteTransformator.xtend:54]:
```suggestion
		positions.addAll(geoKante.geometry.
```

**Review by pstuecker — CHANGES_REQUESTED** [review]

**Review by pstuecker — APPROVED** [review]

**Pull request merged**